packages/components/src/components/badge/badge.test.ts

```diff
@@ -0,0 +1,116 @@
+import { expect, fixture, html } from '@open-wc/testing';
+import type SdBadge from './badge';
+
+const variants = [
+  {
+    name: 'default',
+    tailwindClasses: 'text-white bg-primary-500 border-white',
+    invertedTailwindClasses: 'text-primary bg-white border-primary'
+  },
+  {
+    name: 'success',
+    tailwindClasses: 'text-white bg-[#367B28] border-white',
+    invertedTailwindClasses: 'text-white bg-[#367B28] border-white'
+  },
+  {
+    name: 'error',
+    tailwindClasses: 'text-white bg-error border-white',
+    invertedTailwindClasses: 'text-white bg-error border-white'
+  }
+];
+
+const sizes = [
+  { name: 'sm', tailwindClasses: 'h-2 min-w-[8px] text-[10px]' },
+  { name: 'md', tailwindClasses: 'h-4 px-[4px] min-w-[16px] text-[10px]' },
+  { name: 'lg', tailwindClasses: 'h-5 px-[5px] min-w-[20px] text-[12px]' }
+];
+
+const baseClasses =
+  'inline-flex items-center text-center justify-center align-middle leading-none whitespace-nowrap border rounded-full select-none font-semibold cursor-[inherit]';
+
+describe('<sd-badge>', () => {
+  let el: SdBadge;
+
+  describe('when provided no parameters', () => {
+    it('should pass accessibility tests with a role of status on the base part.', async () => {
+      el = await fixture<SdBadge>(html` <sd-badge>Badge</sd-badge> `);
+      expect(el.shadowRoot!.querySelector('[part~="base"]')!.getAttribute('role')).to.eq('status');
+      await expect(el).to.be.accessible();
+    });
+
+    it('should have the primary values set correctly', async () => {
+      el = await fixture<SdBadge>(html` <sd-badge>Badge</sd-badge> `);
+      expect(el.variant).to.equal('default');
+      expect(el.size).to.equal('lg');
+      expect(el.inverted).to.equal(false);
+    });
+
+    it('should render the child content provided', async () => {
+      el = await fixture<SdBadge>(html` <sd-badge>Badge</sd-badge> `);
+      expect(el.innerText).to.eq('Badge');
+    });
+
+    it('should render without the overflow indicator', async () => {
+      el = await fixture<SdBadge>(html` <sd-badge>Badge</sd-badge> `);
+      expect(el.shadowRoot!.querySelector('[part~="base"]')).to.exist;
+      expect(el.shadowRoot!.querySelector('[part~="overflow-indicator"]')).have.class('hidden');
+      expect(el.querySelector(`*[slot='overflow-indicator']`)).to.not.exist;
+    });
+
+    it('should default to default styling, with the primary color', async () => {
+      el = await fixture<SdBadge>(html` <sd-badge>Badge</sd-badge> `);
+      const part = el.shadowRoot!.querySelector('[part~="base"]')!;
+      expect(part.classList.value.trim()).to.contain(baseClasses);
+      expect(part.classList.value.trim()).to.contain(variants[0].tailwindClasses);
+    });
```

Comment:
As we have visual testing (somethign Shoelace doesn't have!) I would say you can omit all tests that check for specific TailwindClasses.

In that case you're more testing if cx works, but that's not part of our job.